pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/RecordExtractor.java

```diff
@@ -46,4 +48,16 @@
    * @return The output GenericRow
    */
   GenericRow extract(T from, GenericRow to);
+
+  /**
+   * Converts a field of the given input record. The field value will be converted to either a single value
+   * (string, number, bytebuffer), multi value (Object[]) or a Map.
+   *
+   * Natively Pinot only understands single values and multi values.
+   * Map is useful only if some ingestion transform functions operates on it in the transformation layer.
+   *
+   * @param value the field value to be converted
+   * @return The converted field value
+   */
+  Object convert(@Nullable V value);
```

Comment:
Are we returning `null` for empty array/collection/map? If so, let's add the behavior to the javadoc and annotate the return value as nullable